src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -80,6 +80,13 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
             this.addCustomLocationProperties(site, context.customLocation);
         }
 
+        const isNotFree = context.plan?.sku?.family?.toLowerCase() !== 'u';
```

Comment:
This applies to all three. If the `family` / `tier` are `undefined`, these booleans will evaluate to `true`. Is that intentional?

Reply:
Hmm I am not totally sure, but probably not since I think always on should only be set to on when an app service plan that is not in the free tier is chosen.

Reply:
Why are we excluding the free tier?

Reply:
Based off what happens in Azure the free tier does not allow the allow on feature to be changed, so I wasn't sure if we would enable that automatically if users aren't able to change it. But that's the only reason I was excluding it for now. 

![image](https://user-images.githubusercontent.com/59709511/200943055-5aeebf3f-019e-4ace-ad4f-1ae28e9c3380.png)

Reply:
Maybe I'm totally off but I thought that you _couldn't_ have always-on functions on the free tier, and that was an inherent part of the "agreement" for free tier (i.e. we won't give you any idle CPU time for free).

Reply:
> Maybe I'm totally off but I thought that you couldn't have always-on functions on the free tier, and that was an inherent part of the "agreement" for free tier (i.e. we won't give you any idle CPU time for free).

Correct, in order to enable always-on functions, it has to be running on top of an App Service Plan which isn't available for the free tier.

https://learn.microsoft.com/en-us/azure/azure-functions/dedicated-plan#always-on

Reply:
Is there an error when you try to enable it for a free sku? Or does it just ignore the setting?

If it does ignore it, then I think we might as well just always set it on for anything that's not a consumption plan.

Reply:
This is the error I got when I tried to enable it for a free sku. 

![image](https://user-images.githubusercontent.com/59709511/200957959-d603cbdb-3aa1-424e-8d23-23b292952361.png)